File: src/NovelRT.Java/src/main/java/com/github/novelrt/graphics/RenderObject.java
package com.github.novelrt.graphics;

import com.github.novelrt.WorldObject;

public class RenderObject extends WorldObject {
  protected RenderObject(long handle, boolean isOwned) {
    super(handle, isOwned);
  }

  public void executeObjectBehaviour() {
    executeObjectBehaviourNative(getHandle());
  }

  private native void executeObjectBehaviourNative(long handle);
}

File: src/NovelRT.Java/core/src/main/kotlin/com/github/novelrt/graphics/RenderObject.kt
package com.github.novelrt.graphics

import com.github.novelrt.WorldObject
import com.github.novelrt.fumocement.DisposalMethod
import com.github.novelrt.fumocement.HandleDeleter

sealed class RenderObject(
    handle: Long,
    isOwned: Boolean = true,
    deleter: HandleDeleter,
    disposalMethod: DisposalMethod = DisposalMethod.GARBAGE_COLLECTED
) : WorldObject(handle, isOwned, deleter, disposalMethod) {
    abstract fun executeObjectBehaviour()
}

